### PR TITLE
OSS parts of sys/config/reload/license

### PR DESCRIPTION
**Pull request by raskchanky**
ENT PR https://github.com/hashicorp/vault-enterprise/pull/1908

### CODE REVIEW

**Comment by raskchanky** [comment]
@ncabatoff When you get a moment, can you give this a review, since I think the ENT counterpart is ready to go?

File: website/content/api-docs/system/config-reload.mdx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+# `/sys/config/reload`
+
+The `sys/config/reload` endpoint allows reloading specific parts of Vault's configuration.
+Currently, it only supports reloading license information from files on disk.
+
+| Method | Path                          |
+| :----- | :---------------------------- |
+| `GET`  | `/sys/config/reload/:subsystem` |
```
Comment (ncabatoff) [inline website/content/api-docs/system/config-reload.mdx:14]: It's a POST, not a GET.

File: website/content/api-docs/system/config-reload.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+When the `:subsystem` URL parameter is specified as `license`, Vault re-reads the license file pointed to by the `license_path` configuration option and applies the license
+to Vault. Vault may enable or disable various features when this happens, depending on if the features of the given
+license have changed from the license Vault is currently using.
+
+### Sample Request
+
+```shell-session
+$ curl \
+  --header "X-Vault-Token: ..." \
```
Comment (ncabatoff) [inline website/content/api-docs/system/config-reload.mdx:30]: Need to add `-X POST`

**Review by ncabatoff — APPROVED** [review]
Minor doc fixes requested, otherwise LGTM

File: website/content/api-docs/system/config-reload.mdx (hunk 1 above)
Comment (raskchanky) [reply website/content/api-docs/system/config-reload.mdx:14]: It's actually a `PUT` now that I look at it, but good point - I'll update.

**Review by ncabatoff — APPROVED** [review]

**Pull request merged**